### ISSUE TITLE
[Accessibility] Lacks textual equivalent to <svg> elements in notifications.

### ISSUE THREAD

**Issue by Krituhar** [microsoft/vscode#227494]
- VS Code Version: 1.88.0
- OS Version: MacOS Sonoma (v14.6.1)

**[Issue]**
The "i" icon and "tick" icon that appears when user hover on "Editor Language Status" ({}) control are without textual equivalents. 

**[User Impact]**
Screen reader users will not receive an accurate impression of the contents of these images.

**[Code Reference]**
```
<span class="codicon codicon-info">::before</span>
```

**[Recommendation]**
Ensure images provide informative alternative text. Textual equivalents must be both concise and descriptive.

To add a textual equivalent to an `<svg>` element, add a <title> child to the SVG. Place the textual equivalent in the `<title>`, then add an ID to the `<title>`. Finally, on the` <svg>`, add an `aria-labelledby` attribute and set its value to the ID of the `<title>`
Provide "info" for the i icon and "selected" or similar for the tick icon.

Screenshot of the issue:
![Screenshot 2024-09-03 at 5 17 47 PM](https://github.com/user-attachments/assets/47957f0f-050c-444b-bd09-0e6cacc6d331)

**Comment by meganrogge** [comment]
This seems really low priority

**Comment by jrieken** [comment]
This is uses the standard codicon rendering of `IStatusbarEntry#text`

**Comment by bpasero** [comment]
@jrieken no, this is from rendering a custom hover here:

https://github.com/microsoft/vscode/blob/d0b2168a691be86907a2709f701acea482e8a584/src/vs/workbench/contrib/languageStatus/browser/languageStatus.contribution.ts#L290-L295

If I understand this issue correctly, rendering a codicon alone is not sufficient to convey the information to screen reader users and thus needs ARIA labels. 

@Krituhar can you clarify how this should be done specifically for this case? There is no SVG element involved, rather the icon is renderer with a `:before`:

![Image](https://github.com/user-attachments/assets/e78ea974-f3c1-48ea-9fe5-2e254732672f)

Wouldn't it help if the entire hover had a custom ARIA label that is being read out that includes the severity per item?

**Comment by jrieken** [comment]
> @jrieken no, this is from rendering a custom hover here:

But it used `renderLabelWithIcons`, which I assumed you own. Not?

**Comment by bpasero** [comment]
@jrieken no, I do not recall writing any of it, it seems to have come in via https://github.com/microsoft/vscode/pull/105589

But if we can make `renderLabelWithIcons` and maybe `renderIcon` accessible, we can likely fix it for more cases. I am just not sure how to convert a simple `<span>` element with a Codicon to be screen reader accessible without adding to many new elements around it, or without even having access to the parent node.

**Comment by meganrogge** [comment]
@rperez030 and @jooyoungseo, do you have thoughts here?

On the one hand, if the icon is providing useful info, it would be good to convey that. however, doing so will mean it's more verbose.

**Comment by meganrogge** [comment]
Closing as I've seen no users reporting this as a bug
